### ISSUE TITLE
Cannot get/set timestamps in UTC timezone

### ISSUE THREAD

**Issue by cowwoc** [jdbi/jdbi#553]
According to http://stackoverflow.com/a/14070771/14731 we must use `PreparedStatement.setTimestamp(position, value, Calendar)` in order to save a `Timestamp` in UTC timezone (where the local timezone is not UTC). But, looking at https://github.com/jdbi/jdbi/blob/d3ec32b480ebe9bd143b7e4f6f1be244f51f1818/src/main/java/org/skife/jdbi/v2/TimestampArgument.java, there doesn't seem to be a way for users to specify the desired timezone.

**Comment by stevenschlansker** [comment]
Can you not set your JVM to use UTC?  We would accept a PR to improve this, but I believe the "correct" approach is to set UTC globally and just let jdbi inherit it.

**Comment by cowwoc** [comment]
@stevenschlansker Asking users to change a JVM-wide setting in order to set a parameter-specific timezone seems like the wrong way to go. For one, it won't solve use-cases where users sometimes want to use different timezones for different parameters.

Are you open to using Java8 types in this API? If so, you could add `bind(name, DateTime)` and pull the time-zone associated with the latter under the hood.

**Comment by stevenschlansker** [comment]
We'd happily accept enhancements to improve this.  Core devs are pretty overworked at the moment, but we'll get to it on the backlog eventually if nobody else steps up :)

In general I agree with your perspective, that forcing users to change JVM configuration is nasty, but that's how it works today and in the case of timezone there's so many pitfalls to not using UTC that I would just recommend going that way 100% of the time.

We would love to adopt the Java 8 types, but cannot do so in jdbi v2.  v3 will have support for the new types.

**Comment by cowwoc** [comment]
I need to revise my earlier statement. It turns out that `PreparedStatement.setTimestamp(position, Timestamp)` does not actually modify the timestamp you pass in. It will set the value, unchanged, in the database.

The problem is that `Timestamp.from(Instant)` will convert the value of the `Instant` to the JVM's default timezone. So, if you want to save an `Instant` into the database unchanged, you need to first convert the `Instant` from UTC to local timezone. Then when you pass it into `Timestamp.from(Instant)` it will convert the value back from the local timezone into UTC. `PreparedStatement.setTimestamp(position, Timestamp)` will save the literal value, and away you go.

I am guessing that the `Calendar` argument in `PreparedStatement.setTimestamp(position, Timestamp, Calendar)` would allow you to avoid the intermediate conversion to the local timezone.

I hope this explanation helps others.

**Comment by stevenschlansker** [comment]
Yuck.  Thanks for figuring out that interaction.

**Comment by cowwoc** [comment]
I ended up converting the code to JDBC and discovered that the `Calendar` argument did not help. Upon further investigation I discovered that by default MySQL sets `useLegacyDatetimeCode=true` which does the wrong thing (always uses the client timezone, no matter what).

If you set `useLegacyDatetimeCode=false` then `setTimestamp(position, Timestamp)` will do the right thing (convert from the client timezone to UTC on save, convert from UTC back to the client timezone on load). The `Calendar` version is only there if you want to modify the `Timestamp` before sending it across the wire, but you can do that yourself before constructing the `Timestamp`.

In other words, I no longer think that `Calendar` support is required. Closing this issue.
